src/rad/appdata.cpp

```diff
@@ -1319,16 +1321,41 @@ bool ApplicationData::LoadProject( const wxString &file, bool justGenerate )
 
 		if ( older )
 		{
+			wxLogDebug("Old version: %d.%d -- current %d.%d", fbpVerMajor, fbpVerMajor, m_fbpVerMajor, m_fbpVerMinor);
+
 			if( justGenerate ){
 				wxLogError( wxT( "This project file is out of date.  Update your .fbp before using --generate" ) );
 				return false;
 			}
-			if ( wxYES == wxMessageBox( wxT( "This project file is not of the current version.\n" )
-			                            wxT( "Would you to attempt automatic conversion?\n\n" )
-			                            wxT( "NOTE: This will modify your project file on disk!" ), _( "Old Version" ), wxYES_NO ) )
+
+			wxFileName fname(file);
+			wxString   strMsg, strOldBackupName(fname.GetFullName() + wxT(".bak"));
```

Comment:
This is an extremely tricky way to declare and initialize strOldBackupName, read: at first i had no idea what is going on here. We are doing C++ here, please don't use that C-style comma operator style here, declare that string on a separate line. Also don't align the consecutive declarations please. That direct initialization is also not really necessary, i think copy initialization looks more natural here.